### ISSUE TITLE
Remove dependencies on aged_containers

### ISSUE THREAD

**Issue by scottschurr** [ripple/rippled#3337]
## Issue Description
The collection of aged_containers (see `src/ripple/beast/container`) are STL-like containers that track the relative age of their contents.  The aged_containers are specific to rippled, which means that they are maintained locally.  The fact that they are STL-like carries with it all of the associated complexity.

The amount of complexity and maintenance risk that the aged_containers carries is not justified by the role they fill in the rippled code base.  We should find `std::` (preferable) or `boost::` replacements for all uses of the aged_containers in the rippled code base, even if that means incurring some performance penalties.  Once all uses of the aged_containers have been replaced, then we should remove the aged_container source code.

**Comment by nbougalis** [comment]
Alternatively, we should consider whether submitting them for inclusion in **Boost** makes sense. I know that this would require extra effort on our part, unless someone else is willing to adopt them and push them through Boost review.

**Comment by scottschurr** [comment]
If Mr. V wants to adopt them I'd be fine with that.  But I doubt anyone on this team wants to adopt them thoroughly enough to get them through the Boost review process.  I tried doing that once years ago with a small utility that I wrote (and was not accepted).  The Boost review process can be brutal.

**Comment by intelliot** [comment]
@scottschurr do you still think this is worth doing?

**Comment by scottschurr** [comment]
Yes, removing the `aged_containers` from the code base is still very much worth doing.  They are very complicated.  They have already broken a couple of times as we have moved from one C++ Standard to the next.  Fortunately @HowardHinnant was able to figure out fixes.  They should be replaced with standard boost (or other library) functionality that someone (other than XRP Ledger developers) is maintaining.

The `aged_containers` are technical debt at its finest.

**Comment by scottschurr** [comment]
Case in point: https://github.com/XRPLF/rippled/pull/4486

**Comment by intelliot** [comment]
- there may be a boost option to consider
- may be useful to delete the oldest data without doing an expensive search
- should try to find a solution to the problem that has a maintainer
